xchange-gateio/src/main/java/org/knowm/xchange/gateio/service/GateioMarketDataServiceRaw.java

```diff
@@ -106,4 +105,35 @@ public GateioTradeHistory getBTERTradeHistorySince(
     List<CurrencyPair> currencyPairs = new ArrayList<>(bter.getPairs().getPairs());
     return currencyPairs;
   }
+
+  public List<GateioKline> getKlines(CurrencyPair pair, GateioKlineInterval interval, int limit) throws IOException {
+
```

Comment:
i think if the limit is a negative you should just throw, the user should exception if a value like that is passed in not absorb it and turn it to 1

Reply:
Ok, exception added.